dispatch/static/manager/src/js/pages/Integrations/FBInstantArticlesIntegrationPage.js

```diff
@@ -0,0 +1,262 @@
+        <SelectInput
+          options={options}
+          onChange={e => this.updateFacebookPage(pageMap[e.target.value])} />
+          <AnchorButton
+            onClick={ e => this.props.saveIntegration(this.props.token, 'fb-instant-articles', integration)}
+            intent={Intent.SUCCESS}>
+            Enable Instant Articles
+          </AnchorButton>
+      </FormInput>
+
+    )
+  }
+
+  renderFacebookLogin() {
+    return (
+      <FormInput label='Authenticate with Facebook'>
+        <br />
+        <AnchorButton
+          href={fbLoginURI(this.props.integration.settings.client_id)}
+          intent={Intent.PRIMARY}>Authenticate with Facebook</AnchorButton>
+      </FormInput>
+    )
+  }
+
+  renderFacebookInstantArticles(integration) {
+
+    let settings = integration.settings || {}
+    let callback = integration.callback || {}
+
+    if (settings.page_configured) {
+      return (
+        <FormInput label={`Instant Articles is enabled for ${settings.page_name}`}>
+          <br />
+          <AnchorButton
+            onClick={e => this.props.deleteIntegration(this.props.token, 'fb-instant-articles')}
+            intent={Intent.DANGER}>
+            Remove integration
+          </AnchorButton>
+        </FormInput>
+      )
+    }
+
+    if (settings.client_configured && !this.state.editMode) {
+      return (
+
+        <div>
+          <FormInput label='Facebook App ID'>
+            <TextInput
+              disabled={true}
+              value={settings.client_id}
+              fill={true} />
+
+            <AnchorButton onClick={e => this.enterEditMode()}>
+              Change app information
+            </AnchorButton>
+          </FormInput>
+
+          {callback.pages ? this.renderFacebookPages(integration) : this.renderFacebookLogin()}
+
+        </div>
+      )
+    }
+
+    let cancelButton = (
+      <AnchorButton
+        onClick={ e => this.exitEditMode()}>
+        Cancel
+      </AnchorButton>
+    )
+
+    return (
+      <div>
+        <FormInput label='Facebook App ID'>
+          <TextInput
+            placeholder='App ID'
+            value={settings.client_id || ''}
+            fill={true}
+            onChange={ e => this.updateSettings({ client_id: e.target.value }) } />
+        </FormInput>
+
+        <FormInput label='Facebook App Secret'>
+          <TextInput
+            placeholder='App Secret'
+            value={settings.client_secret || ''}
+            fill={true}
+            onChange={ e => this.updateSettings({ client_secret: e.target.value }) } />
+        </FormInput>
+
+        <FormInput>
+          <AnchorButton
+            intent={Intent.SUCCESS}
+            onClick={ e => this.saveIntegration(integration)}>
+            Save settings
+          </AnchorButton>
+          {this.state.editMode ? cancelButton : null}
+        </FormInput>
+      </div>
+    )
+
+  }
+
+  render() {
+
+    let integration = this.props.integration
```

Comment:
is there a dispatch styleguide on let vs const? 80% of the `let`s i've seen here are effectively immutable and could be marked as such if you were so inclined